lib/publish.js

```diff
@@ -1,167 +1,105 @@
 'use strict'
 
-const BB = require('bluebird')
-
-const cacache = require('cacache')
-const figgyPudding = require('figgy-pudding')
-const libpub = require('libnpmpublish').publish
-const libunpub = require('libnpmpublish').unpublish
-const lifecycle = BB.promisify(require('./utils/lifecycle.js'))
+const util = require('util')
 const log = require('npmlog')
-const npa = require('npm-package-arg')
-const npmConfig = require('./config/figgy-config.js')
-const output = require('./utils/output.js')
-const otplease = require('./utils/otplease.js')
-const pack = require('./pack')
-const { tarball, extract } = require('pacote')
-const path = require('path')
-const readFileAsync = BB.promisify(require('graceful-fs').readFile)
-const readJson = BB.promisify(require('read-package-json'))
 const semver = require('semver')
-const statAsync = BB.promisify(require('graceful-fs').stat)
+const pack = require('libnpmpack')
+const libpub = require('libnpmpublish').publish
+const runScript = require('@npmcli/run-script')
+
+const npm = require('./npm')
+const output = require('./utils/output')
+const otplease = require('./utils/otplease')
+const { getContents, logTar } = require('./utils/tar')
 
-publish.usage = 'npm publish [<tarball>|<folder>] [--tag <tag>] [--access <public|restricted>] [--dry-run]' +
+const readJson = util.promisify(require('read-package-json'))
+
+publish.usage = 'npm publish [<folder>] [--tag <tag>] [--access <public|restricted>] [--dry-run]' +
                 "\n\nPublishes '.' if no argument supplied" +
                 '\n\nSets tag `latest` if no --tag specified'
 
-publish.completion = function (opts, cb) {
-  // publish can complete to a folder with a package.json
-  // or a tarball, or a tarball url.
-  // for now, not yet implemented.
-  return cb()
-}
-
-const PublishConfig = figgyPudding({
+const publishConfig = () => ({
   dryRun: 'dry-run',
-  'dry-run': { default: false },
-  force: { default: false },
-  json: { default: false },
-  Promise: { default: () => Promise },
-  tag: { default: 'latest' },
-  tmp: {}
+  tag: 'defaultTag',
+  'dry-run': false,
+  defaultTag: 'latest',
+  json: false,
+  tmp: {},
+  ...npm.flatOptions
```

Comment:
The aliasing is weird here, does `libnpmpublish` or `libnpmpack` do something with those?

Any reason you can't just pass `npm.flatOptions` in as the opts?